_search-plugins/ubi/schemas.md

```diff
@@ -0,0 +1,219 @@
+- The **Search Client**, if separate from the **UBI Client**, forwards the indexed [`query_id`](#query_id) to the **UBI Client**.
+    Even though the the roles of *search* and *UBI event indexing* are separated on this diagram, many implementations can use the same OpenSearch client instance for both roles of searching and index writing. 
+    (6 in the diagram above).
+    {: .note} 
+- The **UBI Client** then indexes all user events with the specified [`query_id`](#query_id) until a new search is performed. At this time, a new `query_id` is generated by **User Behavior Insights** plugin and passed back to the **UBI Client**.
+- If the **UBI Client** interacts with a result **object**, such as during an add to cart event, then the [`object_id`](#object_id), `add_to_cart` [`action_name`](#action_name) and `query_id` are all indexed together, signaling the causal link between the *search* and the *object*.
+ (8 and 9, diagram above)
+
+
+
+## UBI stores
+
+There are two separate stores involved in supporting UBI data collection:
+
+### **UBI queries**
+
+All underlying query information and results ([`object_id`](#object_id)'s) are stored in the `ubi_queries` index, and remain largely invisible in the background.
+
+
+**UBI Queries** [schema](https://github.com/opensearch-project/user-behavior-insights/tree/main/src/main/resources/queries-mapping.json)
+
+The `ubi_queries` index, has the following fields:
+
+- `timestamp` (events and queries): A UNIX timestamp of when the query was received.
+
+- `query_id` (events and queries): A unique ID of the query provided by the client or generated automatically. Different queries with the same text generate different `query_id` values. 
+	
+- `client_id` (events and queries): A user/client ID provided by the client application.
+
+- `query_response_objects_ids` (queries):  An array of object IDs. An ID can have the same value as the `_id` but it is meant to be the externally valid ID of a document, item, or product.
+
+Since UBI manages the `ubi_queries` index, you should never have to write directly to this index (except for importing data).
+
+### **UBI events**
+
+The client side directly indexes events to the `ubi_events` index, linking the event [`action_name`](#action_name), objects (each with an [`object_id`](#object_id)), and queries (each with a [`query_id`](#query_id)) along with any other important event information.
+Because this schema is dynamic, you can add any new fields and structures (such as user information or geo-location information) that are not in the current **UBI Events** [schema](https://github.com/opensearch-project/user-behavior-insights/tree/main/src/main/resources/events-mapping.json) at index time.
+
+Below are the defined fields within the `ubi_events` index:
+
+ <p id="application"> </p>
```

Comment:
I would strongly suggest getting rid of HTML here, if there is no link to this section. HTML makes it harder to refactor documentation.

Reply:
The reason we went with this method of linking content was because the nested structure of the events index and the interdependencies of the nested fields between the two query and event indices.  We'd prefer to leave it this way, at least until we get more real-world feedback on the schema.